### DISCUSSION TITLE
Connection

### DISCUSSION THREAD

**Discussion by George19782** [PTriebold/MotomanConnector#3] [Q&A] [answered]
Hey, 

I find it very nice documentated. Hats off! I am considering this idea so I was curious in your findings because you have tested it. I was interrested in how well the communication was? Were there any issues that needs to look at. I am considering this idea so I was curious in your findings

**Comment by PTriebold** [comment, marked as answer]
Hi,
I'm not sure what you mean by how well the communication was.
With the robots that I tested it with, there were no communication problems apart from the getCoordinates function which crashes the DX controller.

**Reply by George19782** [reply]
Aah Thanks, Nice to know. I am trying to communicate an Robot from an PC. One of the methods that is described in the manuals from Yaskawa is the Ethernet Server. This looked like a sutaible option. However I was not quite sure how to make this possible. I kept searching for an week and today I found your solution. So therefore I contacted you because in my project the communication between the PC and Robot needs to be stable. This means no unexpected connection lost. As far as your answer tells me this isn't the problem. Thanks again!

**Reply by PTriebold** [reply]
In my experience, there were no noteworthy connection losses happening. During testing we ran into a rare problem of the DX200 controller locking up after trying to many failed commands. But this was probably due to an outdated firmware on our controller, since it never happened on the newer YRC1000. 

The connection will be dropped, if the ethernet connection fails, i.e. your ethernet cable falls out, or if the Python program crashes. I would recommend you test your python script extensively and try all the commands you send to the controller before deployment. But for development you should not run into any problems.

**Reply by PTriebold** [reply]
Keep in mind, that there are almost no checks if the arguments you send with the commands are possible! Meaning, there is nothing stopping you from sending unreasonable angles in a move command to the Robot. But unless it was intentionally disabled on your controller, the robot does a rudimentary sanity check before comitting to the move.
